R/checkers.R

```diff
@@ -45,3 +62,258 @@ check_elements_non_neg <- function(x, arg = "x", call = rlang::caller_env()) {
   }
   invisible()
 }
+
+#' Assert that there is no missignness in a particular vector
+#'
+#' @param x the vector to check
+#' @param arg the name of the vector to check
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL, invisibly
+assert_non_missingness <- function(x, arg = "x",
+                                   call = rlang::caller_env(),
+                                   add_err_msg = "") {
+  if (checkmate::anyMissing(x)) {
+    cli::cli_abort(
+      c("{.arg {arg}} has missing values", add_err_msg,
+        "!" = "All elements of{.arg {arg}} should be present",
+        "i" = "Element(s) {.val {which(is.na(x))}} are missing"
+      ),
+      class = "wwinference_input_data_error",
+      call = call
+    )
+  }
+  invisible()
+}
+
+#' Check that there are no repeated elements in the vector
+#'
+#' @description
+#' This function  checks that the elements of a vector are
+#' not repeated.
+#'
+#' @param x the vector to check
+#' @param arg the name of the vector to check
+#' @param call Calling environment to be passed to [cli::cli_abort()] for
+#' traceback.
+#' @param add_err_msg string containing an additional error message,
+#' default is the empty string (`""`)
+#'
+#' @return NULL, invisibly
+assert_no_repeated_elements <- function(x, arg = "x",
+                                        call = rlang::caller_env(),
+                                        add_err_msg = "") {
+  duplicates <- duplicated(x)
+  if (any(duplicates)) {
+    cli::cli_abort(
+      c("{.arg {arg}} has more than one element", add_err_msg,
+        "i" = "Multiple {.arg {arg}} are not currently supported.",
+        "!" = "Duplicate element(s) index: {.val {which(duplicates)}}"
+      ),
+      call = call,
+      class = "wwinference_input_data_error"
+    )
+  }
+  invisible()
+}
+
+
+
+#' Assert that the arguments are either count type or characters
+#' @description
+#' This is for unique identifiers of groupings, which we will allow to
+#' either be character or integers.
+#'
+#'
+#' @param x Object with type checking
+#' @param arg Name of the argument supplying the object
+#' @param call Calling environment to be passed to [cli::cli_abort()]
+#'
+#' @return NULL, invisibly
+assert_int_or_char <- function(x, arg = "x", call = rlang::caller_env()) {
+  # Check if its a character, if it is, check passes. If not,
+  # check if its an integer
+  int_or_char_check_result <- is.character(x)
+  if (!int_or_char_check_result) {
+    int_or_char_check_result <- checkmate::check_integerish(x)
+  }
+
+  if (!int_or_char_check_result) {
+    throw_type_error(
+      object = x,
+      arg_name = arg,
+      expected_type = "integer or character",
+      call = call
+    )
+  }
```

Comment:
In the future we can leverage the `checkmate::check_` string output to give a more verbose failure message, but that's a separate PR